core/common/src/LocalDateRange.kt

```diff
@@ -0,0 +1,272 @@
+     * Uses the range operator notation if the progression is increasing, and `downTo` if it is decreasing.
+     * The step is referenced in days.
+     */
+    override fun toString(): String = if (longProgression.step > 0) "$first..$last step ${longProgression.step}D" else "$first downTo $last step ${longProgression.step}D"
+
+    /**
+     * Returns the number of dates in the progression.
+     * Returns [Int.MAX_VALUE] if the number of dates overflows [Int]
+     */
+    override val size: Int
+        get() = longProgression.size
+
+    /**
+     * Returns true iff every element in [elements] is a member of the progression.
+     */
+    override fun containsAll(elements: Collection<LocalDate>): Boolean = elements.all(::contains)
+
+    /**
+     * Returns true iff [element] is a member of the progression.
+     */
+    override fun contains(element: LocalDate): Boolean = longProgression.contains(element.toEpochDaysLong())
+
+    override fun equals(other: Any?): Boolean = other is LocalDateProgression && longProgression == other.longProgression
+
+    override fun hashCode(): Int = longProgression.hashCode()
+
+    public companion object {
+        internal fun fromClosedRange(
+            rangeStart: LocalDate,
+            rangeEnd: LocalDate,
+            stepValue: Long,
+            stepUnit: DateTimeUnit.DayBased
+        ): LocalDateProgression = LocalDateProgression(rangeStart, rangeEnd, safeMultiplyOrClamp(stepValue, stepUnit.days.toLong()))
+    }
+}
+
+/**
+ * A range of values of type [LocalDate].
+ *
+ * @sample kotlinx.datetime.test.samples.LocalDateRangeSamples.simpleRangeCreation
+ */
+public class LocalDateRange(start: LocalDate, endInclusive: LocalDate) : LocalDateProgression(start, endInclusive, 1), ClosedRange<LocalDate>, OpenEndRange<LocalDate> {
+    /**
+     * Returns the lower bound of the range, inclusive.
+     */
+    override val start: LocalDate get() = first
+
+    /**
+     * Returns the upper bound of the range, inclusive.
+     */
+    override val endInclusive: LocalDate get() = last
+
+    /**
+     * Returns the upper bound of the range, exclusive.
+     */
+    @Deprecated(
+        "This throws an exception if the exclusive end if not inside " +
+                "the platform-specific boundaries for LocalDate. " +
+                "The 'endInclusive' property does not throw and should be preferred.",
+        level = DeprecationLevel.WARNING
+    )
+    override val endExclusive: LocalDate get(){
+        if (last == LocalDate.MAX) error("Cannot return the exclusive upper bound of a range that includes LocalDate.MAX.")
+        return endInclusive.plus(1, DateTimeUnit.DAY)
+    }
+
+    /**
+     * Returns true iff [value] is contained within the range.
+     * i.e. [value] is between [start] and [endInclusive].
+     */
+    @Suppress("ConvertTwoComparisonsToRangeCheck")
+    override fun contains(value: LocalDate): Boolean = first <= value && value <= last
```

Comment:
There's a compiler warning that's actually helpful: `The corresponding parameter in the supertype 'LocalDateProgression' is named 'element'. This may cause problems when calling this function with named arguments.` I agree, let's rename this to `contains(element)`.

Reply:
I saw that earlier. Unfortunately `ClosedRange<T>` and `OpenEndRange<T>` both define the method `public operator fun contains(value: T): Boolean` while `Collection<E>` defines `public operator fun contains(element: E)`. `LocalDateRange` implements all three interfaces. They are all implemented with a single method, so whatever name one chooses for the parameter in that method is going to generate that warning for at least one of the parent interfaces.

Reply:
In the standard library, we chose to have the parameter named `value` both in range's and progression's `contains`